Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -930,20 +944,28 @@ DirectoryBlobPartition::DirectoryBlobPartition(
   BuildFST(std::move(nodes), new_fst_address);
 }
 
+DirectoryBlobPartition::DirectoryBlobPartition(const DirectoryBlobPartition& rhs)
+    : m_contents(rhs.m_contents), m_disc_header(rhs.m_disc_header), m_key(rhs.m_key),
+      m_root_directory(rhs.m_root_directory), m_is_wii(rhs.m_is_wii),
+      m_address_shift(rhs.m_address_shift), m_data_size(rhs.m_data_size),
+      m_wrapped_partition(rhs.m_wrapped_partition)
+{
+}
+
 void DirectoryBlobPartition::SetDiscHeaderFromFile(const std::string& boot_bin_path)
 {
   m_disc_header.resize(DISCHEADER_SIZE);
   if (ReadFileToVector(boot_bin_path, &m_disc_header) < 0x20)
     ERROR_LOG_FMT(DISCIO, "{} doesn't exist or is too small", boot_bin_path);
 
-  m_contents.AddReference(DISCHEADER_ADDRESS, m_disc_header);
+  m_contents.Add(DISCHEADER_ADDRESS, std::move(m_disc_header));
```

Comment:
`m_disc_header` is used later. You can't move it.

Like I mentioned before, I would recommend running the code to see that it works.

Reply:
I have been and I'm not sure what I'm missing because I couldn't get your prior comment about m_wrapped_volume to crash either, I must be doing something wrong there.

Reply:
Is the game you're launching is in the directory format? (i.e. is it extracted?)

Reply:
Yes but it's just now dawning on me that I might be looking at Wii-specific code, aren't I.

Reply:
Well, there's the fix but I wouldn't say it's sufficiently tested yet, just running out of time this morning so I'm getting this in while I can and I'll roll back around to test more tonight/tomorrow/when time allows next.

Reply:
> Yes but it's just now dawning on me that I might be looking at Wii-specific code, aren't I.

Oh yes, that's a good point. I wasn't thinking of that before, but that's entirely true, both of these two problems would only happen with Wii discs.

Reply:
Yeah - the only ISOs I have on this drive are a small handful of GCs for testing, my Wii rips are in the drive I use for my steam deck, I'll have to grab one of those when I'm back at my workstation and give that a try.